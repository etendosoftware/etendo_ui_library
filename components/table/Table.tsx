import React from 'react';
import {
  FlatList,
  Text,
  View,
  TouchableOpacity,
  Image,
  ScrollView,
} from 'react-native';
import {GREEN} from '../../styles/colors';
import {Input} from '../../components/input';
import {TableProps} from './Table.types';
import {styles} from './Table.styles';
import TableHeaders from './components/TableHeaders';
import {
  removeLastBorderCell,
  removeHeaderBorder,
  paintOddRows,
} from '../../helpers/utilTable';

const MovementTable = ({addedItems, config, passDataToParent}: TableProps) => {
  const getIcon = (check: boolean) => {
    if (check) {
      return (
        <Image
          style={{height: 14, width: 14}}
          source={require('../../assets/images/icons/checkTrue.png')}
        />
      );
    } else {
      return (
        <Image
          style={{height: 14, width: 14}}
          source={require('../../assets/images/icons/checkFalse.png')}
        />
      );
    }
  };

  const switchColumn = (col: any, item: any, index: number) => {
    if (col.type === 'cellText') {
      return (
        <ScrollView
          nestedScrollEnabled={true}
          contentContainerStyle={styles.scrollCell}>
          <Text style={styles.cellText}>{item[col.key]}</Text>
        </ScrollView>
      );
    }
    if (col.type === 'cellOnlyRead') {
      return (
        <TouchableOpacity
          activeOpacity={0.6}
          style={styles.centerItem}
          onPress={() => {
            passDataToParent({
              typeColumnPress: col.type,
              key: col.key,
              indexCurrent: index,
            });
          }}>
          <Image
            style={{height: 14, width: 11}}
            source={require('../../assets/images/icons/document.png')}
          />
        </TouchableOpacity>
      );
    }
    if (col.type === 'cellEdit') {
      return (
        <TouchableOpacity
          activeOpacity={0.6}
          style={styles.centerItem}
          onPress={() => {
            passDataToParent({
              typeColumnPress: col.type,
              key: col.key,
              indexCurrent: index,
            });
          }}>
          <Image
            style={[
              {height: 13, width: 13},
              item[col.key] ? {tintColor: GREEN} : {},
            ]}
            source={require('../../assets/images/icons/edit.png')}
          />
        </TouchableOpacity>
      );
    }
    if (col.type === 'cellTextInput') {
      return (
        <Input
          centerText={true}
          value={item[col.key]}
          typeField={'textPressable'}
          onPress={() =>
            passDataToParent({
              typeColumnPress: col.type,
              key: col.key,
              indexCurrent: index,
            })
          }
        />
      );
    }
    if (col.type === 'cellCheck') {
      return (
        <TouchableOpacity
          activeOpacity={0.6}
          style={styles.centerItem}
          onPress={() => {
            passDataToParent({
              typeColumnPress: col.type,
              key: col.key,
              indexCurrent: index,
            });
          }}>
          {(() => {
            getIcon(item[col.key]);
          })()}
        </TouchableOpacity>
      );
    }
  };

  const renderItem = (item: any, index: number, config: any) => {
    return (
      <View
        style={[styles.row, {height: config.heightRow}, paintOddRows(index)]}>
        {config.columns.map((col: any, colIndex: number) => {
          return (
            <View
              style={[
                styles.cell,
                {width: col.width},
                removeLastBorderCell(colIndex, config.columns.length - 1),
              ]}
              key={'movementTable' + colIndex}>
              {switchColumn(col, item, index)}
            </View>
          );
        })}
      </View>
    );
  };
  return (
    <>
      <View style={[styles.container, removeHeaderBorder(config.title)]}>
        <TableHeaders config={config} />
        <FlatList
          data={addedItems}
          renderItem={item => renderItem(item.item, item.index, config)}
          keyExtractor={(item: any, index: number) => 'Table: ' + index}
        />
      </View>
      {!addedItems.length && (
        <View style={[styles.placeholderContainer, {height: config.heightRow}]}>
          <Text style={[styles.titleText]}>No hay resultados.</Text>
        </View>
      )}
    </>
  );
};

export default MovementTable;
