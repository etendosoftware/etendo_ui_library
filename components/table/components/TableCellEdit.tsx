import React from 'react';
import {View} from 'react-native';
import {styles} from '../Table.styles';
import {TableCellProps} from '../Table.types';
import {EditIcon} from '../../../assets/images/icons/EditIcon';

const TableCellEdit = ({isWeb}: TableCellProps) => {
  return (
    <View style={styles.centerItem}>
      <EditIcon style={{height: 18, width: 18}} />
    </View>
  );
};

export default TableCellEdit;
