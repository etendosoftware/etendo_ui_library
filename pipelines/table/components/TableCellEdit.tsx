import React from 'react';
import {Image, Text, View} from 'react-native';
import {styles} from '../Table.styles';
import {TableCellProps} from '../Table.types';
import {editImage} from '../../../assets/images';

const TableCellEdit = ({label}: TableCellProps) => {
  return (
    <View style={styles.centerItem}>
      <Image style={styles.imgEdit} source={editImage} />
      <Text style={styles.cellTextEdit}>{label}</Text>
    </View>
  );
};

export default TableCellEdit;
