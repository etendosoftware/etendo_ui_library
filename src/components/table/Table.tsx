import React from 'react';
import {
  FlatList,
  Image,
  Pressable,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { Actions, Columns, TableProps } from './Table.types';
import { styles } from './Table.styles';
import TableHeaders from './components/TableHeaders';
import TableCell from './components/TableCell';
import {
  isDeviceTablet,
  paintOddRows,
  removeHeaderBorder,
} from '../../helpers/table_utils';
import { SkeletonRowTable } from '../secondaryComponents';
import { NEUTRAL_300 } from '../../styles/colors';

const Table = ({
  data,
  columns,
  title,
  onRowPress,
  tableHeight,
  isLoading,
  pageSize,
  textEmptyTable,
  commentEmptyTable,
}: TableProps) => {
  const findPrimaryId = (col: Columns[], indexRow: number) => {
    let primary: string = '';
    if (indexRow >= 0) {
      for (const element of col) {
        if (element.primary === true) {
          if (element.key) {
            primary = data[indexRow][element.key];
            break;
          }
        }
      }
    }
    return primary;
  };

  const RenderItem = (item: any, index: number) => {
    return (
      <Pressable
        onPress={() => {
          onRowPress(findPrimaryId(columns, index));
        }}
        style={[styles.row, paintOddRows(index)]}>
        {columns.map((col: Columns, colIndex: number) => {
          return (
            col.visible && (
              <View
                style={[styles.cell, col.cellStyle, { width: col.width }]}
                key={'movementTable' + colIndex}>
                {col.components ? (
                  col.components?.map(
                    (itemAction: Actions, actionIndex: number) => {
                      return (
                        <TouchableOpacity
                          style={styles.cellEditContainer}
                          onPress={() =>
                            itemAction.onAction(findPrimaryId(columns, index))
                          }
                          key={'tableCellCustom' + actionIndex}>
                          {itemAction.component}
                        </TouchableOpacity>
                      );
                    },
                  )
                ) : (
                  <TableCell
                    label={col.key ? item[col.key] : ''}
                    key={'tableCell' + index}
                  />
                )}
              </View>
            )
          );
        })}
      </Pressable>
    );
  };

  const RenderSkeleton = (item: any, index: number) => {
    return (
      <Pressable
        onPress={() => {
          onRowPress(findPrimaryId(columns, index));
        }}
        style={[styles.row, paintOddRows(index)]}>
        {columns.map((col: Columns, colIndex: number) => {
          return (
            col.visible && (
              <SkeletonRowTable
                key={colIndex}
                width={col.width!}
                color={NEUTRAL_300}
                indexRow={index}
                indexColumn={colIndex}
              />
            )
          );
        })}
      </Pressable>
    );
  };

  const EmptyState = () => {
    const image = require('../../assets/images/components/empty-state-table.png');
    return (
      <View style={styles.emptyStateConteiner}>
        <Image
          style={
            isDeviceTablet
              ? { width: '50%', height: '70%' }
              : { width: '100%', height: '50%' }
          }
          source={image}
          resizeMode="stretch"
        />
        <Text style={styles.emptyTextTitle}>{textEmptyTable}</Text>
        <Text style={styles.emptyTextSubtitle}>{commentEmptyTable}</Text>
      </View>
    );
  };

  return (
    <>
      <View
        style={[
          styles.container,
          removeHeaderBorder(title),
          { height: tableHeight },
        ]}>
        <TableHeaders title={title} columns={columns} isLoading={!isLoading} />

        <FlatList
          data={!isLoading ? data : Array(pageSize).fill({})}
          showsVerticalScrollIndicator={false}
          showsHorizontalScrollIndicator={false}
          renderItem={item =>
            isLoading
              ? RenderSkeleton(item.item, item.index)
              : RenderItem(item.item, item.index)
          }
          ListEmptyComponent={EmptyState}
          keyExtractor={(_item: any, index: number) => 'Table: ' + index}
        />
      </View>
    </>
  );
};

export default Table;
