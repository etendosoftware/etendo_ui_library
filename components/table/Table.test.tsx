import React from 'react';
import renderer from 'react-test-renderer';
import TableCellEdit from './components/TableCellEdit';
import Table from './Table';

const onPress = (id:string) => {
  console.log(id)
}

describe('Running Test for Table', () => {
  it('Check Table Disabled', () => {
    const tree = renderer
      .create(
        <Table
          onRowPress={primary => {
            console.log(primary);
          }}
          heightRow={40}
          data={[
            {
              searchKey: '09123 ZEL',
              productName: 'Beer 200ml Cusque単a',
              description: '',
              expiration: '20/11/2023',
              lot: '00001',
              movementQuantity: 120,
              move: false,
            },
            {
              searchKey: '09124 ZEL',
              productName: 'Beer 200ml Cusque単a',
              description: '',
              expiration: '20/11/2023',
              lot: '00002',
              movementQuantity: 120,
              move: false,
            },
            {
              searchKey: '09123 ZEL',
              productName: 'Beer 200ml Cusque単a',
              description: '',
              expiration: '20/11/2023',
              lot: '00001',
              movementQuantity: 120,
              move: false,
            },
            {
              searchKey: '09124 ZEL',
              productName: 'Beer 200ml Cusque単a',
              description: '',
              expiration: '20/11/2023',
              lot: '00002',
              movementQuantity: 120,
              move: false,
            },
          ]}
          columns={[
            {
              primary: false,
              displayKey: 'movementQuantity',
              visible: true,
              key: 'quantity',
              width: '25%',
              label: 'Quantity',
            },
            {
              primary: false,
              displayKey: 'lot',
              visible: true,
              key: 'lot',
              width: '25%',
              label: 'Lot',
            },
            {
              primary: false,
              displayKey: 'searchKey',
              visible: true,
              key: 'searchKey',
              width: '25%',
              label: 'Code',
            },
            {
              primary: false,
              displayKey: 'searchKey',
              visible: true,
              key: 'lot',
              width: '25%',
              label: 'Edit',
              actions: [
                {
                  component: <TableCellEdit label={'EDIT'} />,
                  onAction: onPress,
                },
              ],
            },
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
