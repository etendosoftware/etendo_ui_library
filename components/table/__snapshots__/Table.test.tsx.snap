// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Table Check Table Disabled 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#E5E5E5",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "borderWidth": 1,
        "width": "100%",
      },
      Object {
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 0,
        "borderTopWidth": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#EFF1F7",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#E5E5E5",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "justifyContent": "center",
            "paddingVertical": 2,
          },
          Object {
            "height": 40,
            "width": "25%",
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#202452",
              "fontSize": 14,
              "textAlign": "center",
            },
          ]
        }
      >
        Quantity
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#E5E5E5",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "justifyContent": "center",
            "paddingVertical": 2,
          },
          Object {
            "height": 40,
            "width": "25%",
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#202452",
              "fontSize": 14,
              "textAlign": "center",
            },
          ]
        }
      >
        Lot
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#E5E5E5",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "justifyContent": "center",
            "paddingVertical": 2,
          },
          Object {
            "height": 40,
            "width": "25%",
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#202452",
              "fontSize": 14,
              "textAlign": "center",
            },
          ]
        }
      >
        Code
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#E5E5E5",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "justifyContent": "center",
            "paddingVertical": 2,
          },
          Object {
            "height": 40,
            "width": "25%",
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#202452",
              "fontSize": 14,
              "textAlign": "center",
            },
          ]
        }
      >
        Edit
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "description": "",
          "expiration": "20/11/2023",
          "lot": "00001",
          "move": false,
          "movementQuantity": 120,
          "productName": "Beer 200ml Cusque単a",
          "searchKey": "09123 ZEL",
        },
        Object {
          "description": "",
          "expiration": "20/11/2023",
          "lot": "00002",
          "move": false,
          "movementQuantity": 120,
          "productName": "Beer 200ml Cusque単a",
          "searchKey": "09124 ZEL",
        },
        Object {
          "description": "",
          "expiration": "20/11/2023",
          "lot": "00001",
          "move": false,
          "movementQuantity": 120,
          "productName": "Beer 200ml Cusque単a",
          "searchKey": "09123 ZEL",
        },
        Object {
          "description": "",
          "expiration": "20/11/2023",
          "lot": "00002",
          "move": false,
          "movementQuantity": 120,
          "productName": "Beer 200ml Cusque単a",
          "searchKey": "09124 ZEL",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "height": 40,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  120
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  00001
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  09123 ZEL
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/images/icons/edit.png",
                    }
                  }
                  style={
                    Object {
                      "height": 13,
                      "marginRight": 5,
                      "width": 13,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  EDIT
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "height": 40,
              },
              Object {
                "backgroundColor": "#EFF1F7",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  120
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  00002
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  09124 ZEL
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/images/icons/edit.png",
                    }
                  }
                  style={
                    Object {
                      "height": 13,
                      "marginRight": 5,
                      "width": 13,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  EDIT
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "height": 40,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  120
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  00001
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  09123 ZEL
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/images/icons/edit.png",
                    }
                  }
                  style={
                    Object {
                      "height": 13,
                      "marginRight": 5,
                      "width": 13,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  EDIT
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "height": 40,
              },
              Object {
                "backgroundColor": "#EFF1F7",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  120
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  00002
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
              nestedScrollEnabled={true}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  09124 ZEL
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 5,
                },
                Object {
                  "width": "25%",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/images/icons/edit.png",
                    }
                  }
                  style={
                    Object {
                      "height": 13,
                      "marginRight": 5,
                      "width": 13,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#202452",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                >
                  EDIT
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
